scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -522,6 +529,128 @@ impl ReplicatorConsumer {
         Ok(())
     }
 
+    fn delete_row_range_left(&mut self, data: CDCRow<'_>, included: bool) {
+        let (_, _, values) = self.get_common_cdc_row_data(&data);
+        self.left_range_included = included;
+        self.left_range_values = values[self.table_schema.partition_key.len()..]
+            .iter()
+            .map(|el| el.map(|x| x.clone()))
+            .collect();
+    }
+
+    async fn delete_row_range_right(
+        &self,
+        data: CDCRow<'_>,
+        right_included: bool,
+    ) -> anyhow::Result<()> {
```

Comment:
Wow... that turned out to be quite complicated. I think the following observation should help you simplify - this:

```cql
ck1 = a AND ck2 = b AND ck3 > c AND ck3 < d
```

...is equivalent to:

```cql
(ck1, ck2, ck3) > (a, b, c) AND (ck1, ck2, ck3) < (a, b, d)
```

...therefore you can completely forgo using the first form and only use tuple-inequalities.

Reply:
That trick won't work for queries like:
`ck1 = a AND ck2 = b` or `ck1 = a AND ck2 = b AND ck3 > c`

We should also keep in mind that in tables with compact storage queries like
`ck1 = a AND ck2 = null`
are also possible and null values do not work well in tuple-inequalities.

Reply:
Apparently Scylla does not produce any CDC log record for `ck1 = a AND ck2 = null`.

Reply:
Not sure. Perhaps @kbr- will know more about the interaction of compact storage with CDC?

As I wrote, I don't think we should care about compact storage tables in the replicator.